sklearn/utils/validation.py

```diff
@@ -404,9 +404,10 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         if ensure_2d:
             if array.ndim == 1:
                 raise ValueError(
-                    "Got X with X.ndim=1. Reshape your data either using "
-                    "X.reshape(-1, 1) if your data has a single feature or "
-                    "X.reshape(1, -1) if it contains a single sample.")
+                    "Expected 2D array, got 1D array instead: array={}. "
```

Comment:
Even maybe "got 1D array instead:\n{}\n". Otherwise the first row of the array will not be vertically aligned with the other rows